### PR TITLE
Posts: Remove unfinished and disabled bulk editing functionality

### PR DESCRIPTION
**Pull request by tyxla**
A while ago (over 2 years ago) a team started building a bulk edit functionality for post types, but it was never finished (see #18761). It's been disabled in Calypso since then, so it's time to remove it.

#### Changes proposed in this Pull Request

* Posts: Remove unfinished and disabled bulk editing functionality.

#### Testing instructions

* Smoke test posts section in Calypso (listing, filtering, searching).
* Verify all tests pass.

### CODE REVIEW

**Comment by jsnajdr** [comment]
I'm a bit torn about this one: if the Calypso post list is to reach wp-admin parity or even surpass it, then bulk selection and bulk actions are one of the gaps. Did you explore how much work is involved in finishing and releasing this feature?

**Comment by tyxla** [comment]
> I'm a bit torn about this one: if the Calypso post list is to reach wp-admin parity or even surpass it, then bulk selection and bulk actions are one of the gaps. Did you explore how much work is involved in finishing and releasing this feature?

Good question! Yeah, I've thought about that a little bit.

First of all, the post type list in Calypso is currently far away from being even close to having parity with wp-admin. And there hasn't been any movement in that direction for the past couple of years.

Second of all, after giving the existing work a spin, I'm confident that it's pretty far away from finishing. The actual multi selection works right now, but that's the only thing that's working. There is no endpoint for mass post deletion, and there aren't any interfaces built (nor endpoints) for bulk editing multiple posts. 

And finally, this hasn't been touched in over 2 years. I believe we can make the decision to remove it for now, and if we ever decide to bring it back and finish it, it would be relatively straightforward and a quick task, compared to all the work that needs to be done to finish this feature.

File: client/state/ui/post-type-list/reducer.js [hunk 1]
```diff
@@ -20,37 +12,14 @@ import {
...
-			return {
-				...state,
-				selectedPosts: [ ...state.selectedPosts, action.postGlobalId ],
-			};
-
 		case ROUTE_SET:
 			return {
 				...state,
-				isMultiSelectEnabled: false,
 				postIdWithActiveLikesPopover: null,
```
Comment (jsnajdr) [inline client/state/ui/post-type-list/reducer.js:22]: This Likes Popover state should be local to the Post List component. Not a good fit for Redux. It's not related to this PR though, I'm just complaining here about the "put everything into Redux" era 🙂

File: client/state/ui/post-type-list/test/reducer.js [hunk 2]
```diff
@@ -22,8 +20,6 @@ describe( 'reducer', () => {
 		expect( reducer( undefined, {} ) ).to.have.keys( [
```
Comment (jsnajdr) [inline client/state/ui/post-type-list/test/reducer.js:20]: How difficult would it be to rewrite this test to use the Jest `expect` API here and remove the `chai` import?

File: client/state/ui/post-type-list/test/selectors.js [hunk 3]
```diff
@@ -115,101 +109,3 @@ describe( 'isSharePanelOpen', () => {
 		expect( isOpen ).to.be.false;
```
Comment (jsnajdr) [inline client/state/ui/post-type-list/test/selectors.js:109]: Same here: can we remove `chai`?

**Review by jsnajdr — APPROVED** [review]
Looks good! I'd consider rewriting the `chai` assertions to Jest in all the tests that this PR touches.

File: client/state/ui/post-type-list/reducer.js (hunk 1 above)
Comment (tyxla) [reply client/state/ui/post-type-list/reducer.js:22]: Couldn't agree more! I'll make a note to follow-up and clean this area up further.

File: client/state/ui/post-type-list/test/reducer.js (hunk 2 above)
Comment (tyxla) [reply client/state/ui/post-type-list/test/reducer.js:20]: I didn't want to make a gigantic PR out of this. But I'll definitely follow-up in another PR 👍

File: client/blocks/post-item/index.jsx [hunk 4]
```diff
@@ -230,10 +203,8 @@ class PostItem extends React.Component {
 						globalId={ globalId }
 						onClick={ this.clickHandler( 'image' ) }
 					/>
-					{ ! multiSelectEnabled && ! isTypeWpBlock && (
-						<PostActionsEllipsisMenu globalId={ globalId } />
-					) }
-					{ ! multiSelectEnabled && isTypeWpBlock && (
+					{ ! isTypeWpBlock && <PostActionsEllipsisMenu globalId={ globalId } /> }
+					{ isTypeWpBlock && (
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/blocks/post-item/index.jsx:207]: Fixed in #45294

File: client/state/ui/post-type-list/reducer.js (hunk 1 above)
Comment (tyxla) [reply client/state/ui/post-type-list/reducer.js:22]: Followed up with this in #45293.

File: client/state/ui/post-type-list/test/reducer.js (hunk 2 above)
Comment (tyxla) [reply client/state/ui/post-type-list/test/reducer.js:20]: Done as part of #45293.

File: client/state/ui/post-type-list/test/selectors.js (hunk 3 above)
Comment (tyxla) [reply client/state/ui/post-type-list/test/selectors.js:109]: Done as part of #45293.

**Pull request merged**